django/db/models/fields/__init__.py

```diff
@@ -365,14 +365,21 @@ def deconstruct(self):
                     keywords[name] = value
         # Work out path - we shorten it for known Django core fields
         path = "%s.%s" % (self.__class__.__module__, self.__class__.__name__)
-        if path.startswith("django.db.models.fields.related"):
-            path = path.replace("django.db.models.fields.related", "django.db.models")
-        if path.startswith("django.db.models.fields.files"):
-            path = path.replace("django.db.models.fields.files", "django.db.models")
-        if path.startswith("django.db.models.fields.proxy"):
-            path = path.replace("django.db.models.fields.proxy", "django.db.models")
-        if path.startswith("django.db.models.fields"):
-            path = path.replace("django.db.models.fields", "django.db.models")
+        flatten_paths = [
+            "django.db.models.fields.related.base",
+            "django.db.models.fields.related.many_to_one",
+            "django.db.models.fields.related.many_to_many",
+            "django.db.models.fields.related.one_to_one",
+            "django.db.models.fields.related",
+            "django.db.models.fields.files",
+            "django.db.models.fields.proxy",
+            "django.db.models.fields",
+        ]
+        for flatten_path in flatten_paths:
+            if path.startswith(flatten_path):
```

Comment:
maybe remove the `django.db.models.fields` repetition above and use `path.startswith("django.db.models.fields" + flatten_path)` here?